stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -366,6 +389,140 @@ public struct Unsafe${Mutable}BufferPointer<Element>
   public func makeIterator() -> UnsafeBufferPointerIterator<Element> {
     return UnsafeBufferPointerIterator(_position: _position, _end: _end)
   }
+  
+  /// Deallocates the heap block previously allocated at this buffer pointer’s 
+  /// base address. 
+  ///
+  /// This buffer pointer's `baseAddress` must be `nil` or a pointer to a previously 
+  /// allocated memory block. If `baseAddress` is `nil`, this function does nothing.
+  /// Otherwise, the memory must not be initialized or `Pointee` must be a trivial 
+  /// type. This buffer pointer's `count` must be equal to the allocated 
+  /// size of the heap block.
```

Comment:
Can we make it clear that you only use `deallocate()` with the same pointer you received from allocation? That's one of the main clarifying points of this change—would be great to make sure that concept is in the documentation (here and in the other `deallocate()` methods).